src/ripple/app/ledger/Consensus.h

```diff
@@ -0,0 +1,82 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_LEDGER_IMPL_CONSENSUS_H_INCLUDED
+#define RIPPLE_APP_LEDGER_IMPL_CONSENSUS_H_INCLUDED
+
+#include <ripple/app/ledger/LedgerConsensus.h>
+#include <ripple/app/misc/FeeVote.h>
+#include <ripple/app/misc/NetworkOPs.h>
+#include <ripple/app/tx/InboundTransactions.h>
+#include <ripple/app/tx/LocalTxs.h>
+
+#include <beast/cxx14/memory.h> // <memory>
+
+namespace ripple {
+
+/** Implements the consensus process and provides inter-round state. */
+class Consensus
```

Comment:
I don't feel strongly about this, but I'm kinda with Vinnie on this one.  There are five methods in `ConsensusImp` that are not overrides of those in `Consensus`.  Those could simply be private methods.  It's true that they are still "published", but by making them private your intent is clear.

The other things that are hidden are the data members of `ConsensusImp`.  The interesting members are: `beast::Journal`, `NetworkOPs&`, and an `STValidation::pointer`.  It seems to me that these are common enough types that we needn't work hard to hide them.  Your mileage may vary.

There is both a runtime and a mental cost to the number of `virtual`s in the rippled code base.